### ISSUE TITLE
Update distribution

### ISSUE THREAD

**Issue by tpflueger** [socs-dev-env/dosocs2#87]
Could there be a possibility that DoSOCSv2 be released on PyPI? I'm not sure if there would be any benefit, but then we could fully utilize pip.

**Comment by ghost** [comment]
Hi Tyler,

That's a great idea. However, I am not sure about few things. DoSOCSv2 has
dependencies from FOSSology I.e. for the nomos scanner.

Glad that you brought this up. This is a direction worth exploring.

**Comment by ghost** [comment]
@spacetag what's your take on `pip install dosocs2` instead of git clone followed by install

**Comment by tpflueger** [comment]
Within setup.py, there could be a custom command that runs the shell to install nomos. I haven't checked, but I think that would likely be the biggest concern.

Edit: Just tested, and I was able to setup a python egg of the nomos script to be installed and ran with the setup.py.

**Comment by ghost** [comment]
@tpflueger here is how I look at dosocs2. 

## dosocs2 will remain:
1. Agnostic to scanner being used. 
2. Agnostic to databases. 

## Nomos
1. Nomos will be part of FOSSology project
2. Each time the script is called to install nomos, source is pulled and compiled.
3. While it is possible to have a postinstall script to do that, Nomos is not the only scanner.

## Agnostic
1. If DoSOCSv2 is to remain agnostic to scanner, I am assuming that a distribution of DoSOCSv2 should adhere to the same philosophy. Installing a scanner either nomos or other license scanners is a user's choice. 
2. Here is something of interest. https://github.com/DoSOCSv2/DoSOCSv2/blob/master/dosocs2/scannerbase.py

## More scanners
1. https://github.com/fossology/fossology/tree/master/src/monk
2. https://github.com/fossology/fossology/tree/master/src/ninka
3. https://github.com/fossology/fossology/tree/master/src/copyright
4. https://github.com/nexB/scancode-toolkit

in DoSOCSv2/dosoc2/scanners, current implementation is for nomos only. However, more scanners can be used. Having a distribution for one scanner is limiting. Instead having dosocs2 distribution without any scanners is an option. 

## Configuration for new scanners

As soon as a new scanner is installed or available, the ideal location to be looking at is dosocs2.conf. To find the location of current configuration file type `dosocs2 configtest` . This file can either be edited to accommodate new scanner or a`dosocs2 newconfig` can be used create a new configuration file and then edited.

**Comment by ghost** [comment]
@tpflueger I am closing this for now. If you still want to accommodate dosocs2 distribution with nomos, please reopen this issue.

**Comment by tpflueger** [comment]
@UShan89 I can understand the reasoning not to specify what database or scanner to use. Only reason I suggested it because I'm trying to pull in dosocs2 for my CI build. But I think it still would be useful for people to install instead of cloning an entire repo for a command.

**Comment by ghost** [comment]
I agree. I do see your point. Let me see if I can come up with a solution.
